src/mad_spark_alt/evolution/genetic_algorithm.py

```
@@ -523,11 +533,11 @@ def _adapt_mutation_rate(
     ) -> float:
         """Adapt mutation rate based on population state."""
         # Increase mutation if diversity is low
-        if snapshot.diversity_score < 0.3:
-            return min(current_rate * 1.2, 0.5)
+        if snapshot.diversity_score < MODERATE_DIVERSITY_THRESHOLD:
+            return min(current_rate * MUTATION_RATE_INCREASE_FACTOR, MAX_MUTATION_RATE)
         # Decrease mutation if diversity is high
-        elif snapshot.diversity_score > 0.7:
-            return max(current_rate * 0.8, 0.01)
+        elif snapshot.diversity_score > HIGH_DIVERSITY_THRESHOLD:
```

Ruff (0.12.2)

Unnecessary `elif` after `return` statement

Remove unnecessary `elif`

(RET505)